src/sentry/sentry_metrics/indexer/models.py

```diff
@@ -50,3 +58,146 @@ class Meta:
         constraints = [
             models.UniqueConstraint(fields=["string", "organization_id"], name="unique_org_string"),
         ]
+
+    @classmethod
+    def get_many_ids(cls, keys) -> Mapping[str, str]:
```

Comment:
A question about the purpose of these methods: The application (both the product code and the indexer consumer code) will still interact with these through the `StringIndexer` service that returns instances of the `StringIndexer` model, correct?
This as opposed to making the product code interact directly with the cache.

I think going through the service is the right option.

Reply:
+1, there should be only one public interface for (for example) resolving an ID, and I think that should be the service class. Would it make sense to create a new implementation of the service class, and move the caching logic there?

Reply:
@fpacifici @jjbayer sounds good, I'll move the caching logic into the service